### PR TITLE
Added support for non-empty standard values

### PR DESCRIPTION
**Pull request by umbreak**
Implemented using cats data structures for
- NonEmptyList
- NonEmptySet
- NonEmptyVector

Added our own (simple) implementation for `NonEmptyString`

### CODE REVIEW

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Decoder.scala [hunk 1]
```diff
@@ -155,13 +158,17 @@ object Decoder extends PrimitiveDecoderInstances with StandardDecoderInstances w
...
+
   implicit final val graphDecodeString: Decoder[String] = Decoder.instance { c =>
     c.narrow.focus match {
       case Some(lit: Literal) if lit.isString => Right(lit.lexicalForm)
       case _                                  => Left(DecodingError("Unable to decode node as a literal String", c.history))
     }
   }
 
+  implicit final val graphDecodeNonEmptyString: Decoder[NonEmptyString] =
+    graphDecodeString.emap(NonEmptyString(_).toRight(s"Unable to decode node as a empty String"))
```
Comment (bogdanromanx) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Decoder.scala]: `.toRight("Unable to decode node as a NonEmptyString")`

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Decoder.scala [hunk 2]
```diff
@@ -285,6 +292,26 @@ trait StandardDecoderInstances { this: PrimitiveDecoderInstances =>
         case Right(a) => Right(Left(a))
       }
     }
+
+  implicit final def graphDecodeNonEmptySet[A: Decoder: Order]: Decoder[NonEmptySet[A]] =
+    graphDecodeSet[A].emap { set =>
+      set.headOption match {
+        case Some(head) => Right(NonEmptySet(head, SortedSet(set.drop(1).toSeq: _*)))
```
Comment (bogdanromanx) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Decoder.scala]: why a SortedSet? can it be like this?
```
set.headOption match {
  case Some(head) => Right(NonEmptySet(head, set - head))
  ...
}
```

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Encoder.scala [hunk 3]
```diff
@@ -133,6 +135,13 @@ trait StandardEncoderInstances {
         case Right(b) => B(b)
       }
     }
+
+  implicit final def graphEncodeNonEmptySet[A](implicit A: Encoder[A]): Encoder[NonEmptySet[A]] =
+    graphEncodeSet[A].contramap(_.toSortedSet)
```
Comment (bogdanromanx) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Encoder.scala]: do we really need to sort the set?

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/NonEmptyString.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package ch.epfl.bluebrain.nexus.rdf
+
+/**
+  * A simple non empty string implementation
+  *
+  * @param head the head character
+  * @param rest the rest of the string
+  */
+final case class NonEmptyString(head: Char, rest: String) {
```
Comment (bogdanromanx) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/NonEmptyString.scala]: make ctor private

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Decoder.scala (hunk 2 above)
Comment (umbreak) [reply modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Decoder.scala]: `NonEmptySet` implementation is a `SortedSet` implementation

**Pull request merged**